diff --git a/src/muc_light/mod_muc_light.erl b/src/muc_light/mod_muc_light.erl
index e3f9903fe..6e3c8ff53 100644
--- a/src/muc_light/mod_muc_light.erl
+++ b/src/muc_light/mod_muc_light.erl
@@ -30,7 +30,8 @@
 %% For Administration API
 -export([try_to_create_room/3,
          change_room_config/5,
-         delete_room/1]).
+         delete_room/1,
+         equal_occupants/1]).
 
 %% gen_mod callbacks
 -export([start/2, stop/1, config_spec/0, supported_features/0, deps/2]).
diff --git a/src/muc_light/mod_muc_light_commands.erl b/src/muc_light/mod_muc_light_commands.erl
index b7f831f8e..7ccf7c8ef 100644
--- a/src/muc_light/mod_muc_light_commands.erl
+++ b/src/muc_light/mod_muc_light_commands.erl
@@ -239,7 +239,11 @@ delete_room(DomainName, RoomName, Owner) ->
     OwnerJID = jid:binary_to_bare(Owner),
     case muc_light_room_name_to_jid_and_aff(OwnerJID, RoomName, DomainName) of
         {ok, RoomJID, owner} -> mod_muc_light:delete_room(jid:to_lus(RoomJID));
-        {ok, _, _} -> {error, denied, "you can not delete this room"};
+        {ok, RoomJID, _} ->
+            case mod_muc_light:equal_occupants(mod_muc_light_utils:room_jid_to_host_type(RoomJID)) of
+                true -> mod_muc_light:delete_room(jid:to_lus(RoomJID));
+                false -> {error, denied, "you can not delete this room"}
+            end;
         {error, given_user_does_not_occupy_any_room} -> {error, denied, "given user does not occupy this room"};
         {error, not_found} -> {error, not_found, "room does not exist"}
     end.
